libraries/hatch-server-middleware/src/CsrfMiddleware.ts

```diff
@@ -0,0 +1,54 @@
+import {ServerMiddleware} from '@launchtray/hatch-server';
+import {injectable} from '@launchtray/hatch-util';
+import {Application, NextFunction, Response, Request} from 'express';
+import cookie from 'cookie';
+import createHttpError from 'http-errors';
+import {StatusCodes} from 'http-status-codes';
+
+@injectable()
+export default class CsrfMiddleware implements ServerMiddleware {
+  public async register(app: Application) {
+    app.use(isCsrfSafe);
+  }
+}
+
+const isMethodSideEffectSafe = (method: string): boolean => {
+  return ['GET', 'HEAD', 'OPTIONS'].includes(method.toUpperCase());
+};
+
+const isCsrfSafe = (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Is this duplicated in the auth middleware? Can that call this?

Reply:
removed duplication